src/base/bittorrent/filterparserthread.cpp

```diff
@@ -37,255 +42,151 @@
     {
-        IPv4Parser parser;
-        lt::error_code ec;
-
-        if (parser.tryParse(data))
-            address = lt::address_v4(parser.parsed());
+        IPFilterEntry entry;
+
+        // Each line should follow this format:
+        // Some organization:1.0.0.0-1.255.255.255
+        // The "Some organization" part might contain a ':' char itself so we find the last occurrence
+        // Apparently this format wasn't designed for IPv6, so no effort will be made
+        // to parse possible IPv6 literals wrapped in []. These characters can appear in
+        // the "some organization" part too. It isn't worth implementing such complex parsing
+        // for a format that doesn't support IPv6 and there aren't any lists with IPv6 in them.
+        if (const auto colon = line.rfind(':');
+            colon != line.npos)
+            entry.range = line.substr(colon + 1);
         else
-            address = lt::make_address(data, ec);
+            entry.range = line;
 
-        return !ec;
+        return entry;
     }
 
-    const int BUFFER_SIZE = 2 * 1024 * 1024; // 2 MiB
-    const int MAX_LOGGED_ERRORS = 5;
-}
-
-FilterParserThread::FilterParserThread(QObject *parent)
-    : QThread(parent)
-    , m_abort(false)
-{
-}
-
-FilterParserThread::~FilterParserThread()
-{
-    m_abort = true;
-    wait();
-}
+    std::pair<std::string_view, std::string_view> parseRange(std::string_view range)
+    {
+        std::string_view start;
+        std::string_view end;
 
-// Parser for eMule ip filter in DAT format
-int FilterParserThread::parseDATFilterFile()
-{
-    int ruleCount = 0;
-    QFile file(m_filePath);
-    if (!file.exists()) return ruleCount;
+        if (const auto pos  = range.find('-');
+            pos != range.npos)
+        {
+            start = range.substr(0, pos);
+            end = range.substr(pos + 1);
+        }
 
-    if (!file.open(QIODevice::ReadOnly | QIODevice::Text))
-    {
-        LogMsg(tr("I/O Error: Could not open IP filter file in read mode."), Log::CRITICAL);
-        return ruleCount;
+        return std::make_pair(start, end);
     }
 
-    std::vector<char> buffer(BUFFER_SIZE, 0); // seems a bit faster than QVector
-    qint64 bytesRead = 0;
-    int offset = 0;
-    int start = 0;
-    int endOfLine = -1;
-    int nbLine = 0;
-    int parseErrorCount = 0;
-    const auto addLog = [&parseErrorCount](const QString &msg)
-    {
-        if (parseErrorCount <= MAX_LOGGED_ERRORS)
-            LogMsg(msg, Log::CRITICAL);
-    };
-
-    while (true)
+    std::optional<lt::address> parseIPv4Address(std::string_view IPstr)
     {
-        bytesRead = file.read(buffer.data() + offset, BUFFER_SIZE - offset - 1);
-        if (bytesRead < 0)
-            break;
-        const int dataSize = bytesRead + offset;
-        if ((bytesRead == 0) && (dataSize == 0))
-            break;
+        lt::address_v4::bytes_type octets = {};
+        unsigned char octetIndex = 0;
```

Comment:
Declaring a variable as `unsigned char` (and not just `int`) may not be of any use in such a case. Also I would avoid unsigned types unless you sure it really required in some case.